### PR TITLE
Add LIT tests to PR checks

### LINKED ISSUE #62: Add LIT tests to PR check
**Description:**
Add the LIT tests (integration tests) to the Github PR check workflow 

**Suggested Labels:**
Priority/High 

**Suggested Assignees:**
ruvi-d

### LINKED ISSUE #103: Make cbindgen optional
**Description:**
Currently the top level CMake invokes cbindgen to generate the C header file for the Rust runtime library. Since the C header is not mandatory to build nBallerina, calling cbindgen should be moved to an optional CMake target. This change would allow us to build nBallerina from source on platforms without cbindgen installed.

**Suggested Labels:**

Type/Task, Points/1, Priority/Low

**Suggested Assignees:**

ruvi-d 

**Affected Product Version:**

**OS, DB, other environment details and versions:**    

**Steps to reproduce:**

**Related Issues:**

### PR DESCRIPTION
**Pull request by ruvi-d**
## Purpose
* Executes LIT integration tests on pull requests. 
* Makes the use of cbindgen optional during default build target

Resolves #62, Resolves #103

### CODE REVIEW

File: .github/workflows/PRCheck.yml [hunk 1]
```diff
@@ -8,15 +8,46 @@ on:
...
+          curl -L  https://maven.pkg.github.com/ballerina-platform/ballerina-lang/org/ballerinalang/jballerina-tools/2.0.0-Preview2-nballerina-r1/jballerina-tools-2.0.0-Preview2-nballerina-r1.zip -u "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}" -o jballerina-tools-2.0.0-Preview2-nballerina-r1.zip
+          unzip -q jballerina-tools-2.0.0-Preview2-nballerina-r1.zip
+          cd jballerina-tools-2.0.0-Preview2-nballerina-r1/bin
+          echo $(pwd) >> $GITHUB_PATH
+        working-directory: ./build
+
+      - name: Run LIT tests
+        run: | 
+          make check
+        working-directory: ./build
```
Comment (Kishanthan) [inline .github/workflows/PRCheck.yml:53]: we are missing eof line

File: .github/workflows/PRCheck.yml [hunk 2]
```diff
@@ -8,15 +8,46 @@ on:
...
+        run: | 
+          make runtime_header
+        working-directory: ./build
+      
+      - name: Setup Ballerina pack
+        run: |
+          curl -L  https://maven.pkg.github.com/ballerina-platform/ballerina-lang/org/ballerinalang/jballerina-tools/2.0.0-Preview2-nballerina-r1/jballerina-tools-2.0.0-Preview2-nballerina-r1.zip -u "${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}" -o jballerina-tools-2.0.0-Preview2-nballerina-r1.zip
+          unzip -q jballerina-tools-2.0.0-Preview2-nballerina-r1.zip
+          cd jballerina-tools-2.0.0-Preview2-nballerina-r1/bin
+          echo $(pwd) >> $GITHUB_PATH
```
Comment (Kishanthan) [inline .github/workflows/PRCheck.yml:47]: is this added for debug purpose?

File: runtime/CMakeLists.txt [hunk 3]
```diff
@@ -1,12 +1,23 @@
...
+    BUILD_ALWAYS TRUE)
+
+# Add seperate (optional) target to generate header file for runtime lib using cbindgen
+add_custom_target(runtime_header
+  COMMENT "Generate C header file for runtime lib"
+  WORKING_DIRECTORY "${CMAKE_SOURCE_DIR}/runtime"
+  COMMAND cbindgen --config cbindgen.toml --crate BallerinaRT --output include/ballerina_rt.h
+  DEPENDS BallerinaRT)
+
+add_dependencies(runtime_header BallerinaRT)
```
Comment (Kishanthan) [inline runtime/CMakeLists.txt:23]: Is it a general practice in CMake files without eof line?

File: .github/workflows/PRCheck.yml [hunk 4]
```diff
@@ -8,15 +8,46 @@ on:
 
 jobs:
   build:
-
     runs-on: ubuntu-20.04
```
Comment (Kishanthan) [inline .github/workflows/PRCheck.yml:11]: What about windows build? Don't we need the same changes for that?

File: .github/workflows/PRCheck.yml (hunk 2 above)
Comment (ruvi-d) [reply .github/workflows/PRCheck.yml:47]: The `echo path >> $GITHUB_PATH` is the Github workflow way to add to the system PATH variable

File: .github/workflows/PRCheck.yml (hunk 4 above)
Comment (ruvi-d) [reply .github/workflows/PRCheck.yml:11]: We don't have build jobs for Mac or Windows yet. Windows build is anyway not functional at the moment. We should create new issues to setup these build jobs.

File: .github/workflows/PRCheck.yml (hunk 1 above)
Comment (ruvi-d) [reply .github/workflows/PRCheck.yml:53]: will add

File: runtime/CMakeLists.txt (hunk 3 above)
Comment (ruvi-d) [reply runtime/CMakeLists.txt:23]: will add

File: .github/workflows/PRCheck.yml (hunk 2 above)
Comment (Kishanthan) [reply .github/workflows/PRCheck.yml:47]: ack

**Review by Kishanthan — APPROVED** [review]

**Pull request merged**